### ISSUE TITLE
Docker image no longer starts if you set SYNAPSE_SMTP_HOST

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#5430]
As far as I can tell, the `SYNAPSE_SMTP_*` settings were never used, because `email_notifs` was set to `false` in the `homeserver.yaml `used by the docker image. Nevertheless the documentation kinda encourages you to set the `SYNAPSE_SMTP_*` settings. Those that did so will now find that their servers refuse to start, with:

```
RuntimeError: email.password_reset_behaviour is set to 'local' but no public_baseurl is set. This is necessary to generate password reset links
```

**Comment by richvdh** [comment]
The workarounds for now are either:
* don't set the `SYNAPSE_SMTP_*` env vars (will disable password reset on your server)
* use a custom config file via `SYNAPSE_CONFIG_PATH`

**Comment by anoadragon453** [comment]
A quick fix before figuring out the whole Docker mess is just to add a `SYNAPSE_PUBLIC_BASEURL` environment variable.

But we should really figure this out :/

**Comment by richvdh** [comment]
I'm not convinced adding a `SYNAPSE_PUBLIC_BASEURL` is the best approach.

We're going to be telling people to switch to the `SYNAPSE_CONFIG_PATH` approach, so rather than having them add a `SYNAPSE_PUBLIC_BASEURL`, I think we should tell them to switch to a generated file.

The only question is whether we can do something better for people whose working configs have been broken by 1.0, like disabling password resets altogether?

Given the `SYNAPSE_SMTP_` vars are unused in synapse <0.99, maybe we should just remove them, and skip generating the `email` config section?

**Comment by richvdh** [comment]
(incidentally it might be good to do that against the release branch and we can update the docker image on dockerhub.)

**Comment by anoadragon453** [comment]
> Given the `SYNAPSE_SMTP_` vars are unused in synapse <0.99, maybe we should just remove them, and skip generating the email config section?

This sounds best to me, considering, as you say, they were never actually used.

**Comment by hawkowl** [comment]
Because of https://github.com/matrix-org/synapse/issues/5518, I believe we can say "use a proper config" here, and close it?

**Comment by richvdh** [comment]
> Because of #5518, I believe we can say "use a proper config" here, and close it?

As per https://github.com/matrix-org/synapse/issues/5430#issuecomment-504494913: I'd prefer not to break peoples' working configs, and I think we can do better.

**Comment by xobs** [comment]
The `README.md` on https://hub.docker.com/r/matrixdotorg/synapse still documents those variables, so that should first be removed prior to closing this issue.

**Comment by richvdh** [comment]
the readme will be updated at the same time as the docker image is updated on hub.docker.

fixed by #5596
